### PR TITLE
Cover block: Fix gradient overlay (remove black background color)

### LINKED ISSUE #38764: Cover block gradient overlay has a black background color
### Description

When a theme preset gradient is set as an overlay, the overlay [background color is set to black by default Cover block styles](https://github.com/WordPress/gutenberg/blob/4bfd7b3d8a9ff575ccae27ee690ccd7327811574/packages/block-library/src/cover/style.scss#L50).

This causes issues, when a gradient is set with a (semi)transparent color stop and gradient is provided as a theme preset. (An example can be seen in the intro section of the page at https://themedemos.webmandesign.eu/francesca/)

When gradient is user defined in editor, this issue does not occur as inline `background: gradient(here)` style is being applied, which effectively overrides Cover block default styles.

### Step-by-step reproduction instructions

1. Use a theme that contains gradient presets with transparent color stops. Or add such preset to your theme (not FSE theme) following https://developer.wordpress.org/block-editor/how-to-guides/themes/theme-support/#block-gradient-presets instructions.
2. Create a page and add Cover block in its content.
3. Set a background image for the Cover block and set an overlay to use a theme gradient with a transparent color stop.
4. You should see black color instead of the transparent stop in the Cover block overlay both in editor and in front-end.

### Screenshots, screen recording, code snippet

cover-block-transparent-color-stop-gradient-issue

### Environment info

- WordPress 5.9
- No Gutenberg plugin

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

### PR DESCRIPTION
**Pull request by webmandesign**
## Description

Fixes #38764 by setting a transparent background color on a gradient overlay element so transparent gradient stops don't reveal a default black background color of an overlay.

## Types of changes

Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal). 
- [ ] I've updated related schemas if appropriate.

### CODE REVIEW

**Review by glendaviesnz — APPROVED** [review]
Thanks @webmandesign. This fix worked for me. I was able to replicate the issue with a non-block theme by adding the following to `functions.php`:
```php
	add_theme_support(
		'editor-gradient-presets',
		array(
			array(
				'name'     => __( 'Red to transparent', 'test' ),
				'gradient' => 'linear-gradient(135deg, rgba(255,0,0,.8) 0%, rgba(255,0,0,0) 100%)',
				'slug'     => 'red-to-transparent',
			),
		)
	);
```
and the following css:

```css
.has-red-to-transparent-gradient-background {
		background-image: linear-gradient(135deg, rgba(255,0,0,.8) 0%, rgba(255,0,0,0) 100%);
	}
```

Without this patch applied the default black background showed through the transparent part of the gradient instead of the background image:
Screen Shot 2022-02-21 at 4 30 11 PM

With patch applied the image show through as expected in the editor and the frontend:
Screen Shot 2022-02-21 at 4 29 26 PM

I also tested the following combinations on a non-block and blocked based theme and they all still functioned as expected with this patch applied:

- Cover with image showed default block overlay if no overlay color selected
- Cover with no image and solid overlay color selected 
- Cover with image and solid overlay color selected

Given the recent issues with some of the Cover block CSS I think it would be good to get a second set of 👀  and approval on this before we merge it.

File: packages/block-library/src/cover/style.scss
Comment (ramonjd) [inline packages/block-library/src/cover/style.scss:56]: This is working well for me on trunk. Thanks for the patch!

I'm looking at the [deprecations](https://github.com/WordPress/gutenberg/blob/4bfd7b3d8a9ff575ccae27ee690ccd7327811574/packages/block-library/src/cover/deprecated.js#L747) and it looks like `has-background-dim` is the most reliable class to gain specificity here going back to v4. 

Do we need the child combinator (`>`)? For me it carries enough specificity weight without it.

```css
	.has-background-dim.has-background-gradient {
		background-color: transparent;
	}
```

**Review by andrewserong — APPROVED** [review]
Thanks for the fix @webmandesign! This is testing nicely for me, too, it looks like it essentially overwrites the `background-color: $black` rule directly above it for those cases where the `has-background-gradient` class is applied.

I suppose an alternative would be to add an additional chaining to the `:not` rules above, but I quite like this separate rule as it's a bit more readable.

It also didn't look like the child combinator was necessary to me if you want to remove it (it seems to work well either way with nested cover blocks), so I'd be happy with it either way.

LGTM!

File: packages/block-library/src/cover/style.scss
Comment (webmandesign) [reply packages/block-library/src/cover/style.scss:56]: Thanks for spotting the child combinator issue. Indeed, it is not needed so I went ahead and removed it.

**Comment by webmandesign** [comment]
Thanks guys. I have updated the code and removed the child combinator.

**Comment by webmandesign** [comment]
Can I ask also for an opinion on #38761 (with #38762 fix)? Thanks.

**Pull request merged**
